java/tests/com/google/privacy/differentialprivacy/CountTest.java

```diff
@@ -512,4 +513,146 @@ private static void testForBias(
+            .thenAnswer(invocation -> new GaussianNoise().computeConfidenceInterval(
+                    (Long) invocation.getArguments()[0],  (Integer) invocation.getArguments()[1], (Long) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3],  (Double) invocation.getArguments()[4],  (Double) invocation.getArguments()[5]));
+    // Mock the noise mechanism.
+    count =
+            Count.builder()
+                    .epsilon(0.5)
+                    .delta(0.9)
+                    .noise(noise)
+                    .maxPartitionsContributed(15)
+                    .build();
+    count.increment();
+    count.computeResult();
+
+    // The result interval = (-1, 3), but count can't be equal to  0 then it should be clamped to 0.
+    assertThat(count.computeConfidenceInterval(0.152145599)).isEqualTo(ConfidenceInterval.create(0,3));
+  }
+
+  @Test
+  public void computeConfidenceInterval_laplaceTest() {
+    when(noise.computeConfidenceInterval(anyLong(), anyInt(), anyLong(), anyDouble(), anyDouble(), anyDouble()))
+            .thenAnswer(invocation -> new LaplaceNoise().computeConfidenceInterval(
+                    (Long) invocation.getArguments()[0],  (Integer) invocation.getArguments()[1], (Long) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3],  null,  (Double) invocation.getArguments()[5]));
+    // Mock the noise mechanism. Since noise is not Laplace, nor Gaussian, delta will be passed as a value instead of null, in order to pass the checks.
+    count =
+            Count.builder()
+                    .epsilon(0.1)
+                    .noise(noise)
+                    .delta(0.5)
+                    .maxPartitionsContributed(1)
+                    .build();
+    count.incrementBy(10);
+    count.computeResult();
+
+    assertThat(count.computeConfidenceInterval(0.5)).isEqualTo(ConfidenceInterval.create(3,17));
+  }
+
+  @Test
+  public void throwError_Long_whenComputeResultNotCalled() {
+    when(noise.computeConfidenceInterval(anyDouble(), anyInt(), anyDouble(), anyDouble(), anyDouble(), anyDouble()))
+            .thenAnswer(invocation -> new LaplaceNoise().computeConfidenceInterval(
+                    (Long) invocation.getArguments()[0],  (Integer) invocation.getArguments()[1], (Long) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3], null,  (Double) invocation.getArguments()[5]));
+    // Mock the noise mechanism. Since noise is not Laplace, nor Gaussian, delta will be passed as a value instead of null, in order to pass the checks.
+    count = Count.builder()
+            .epsilon(0.1)
+            .noise(noise)
+            .delta(0.5)
+            .maxPartitionsContributed(1)
+            .build();
+    count.incrementBy(10);
+    Exception exception = assertThrows(IllegalStateException.class, () -> {
```

Comment:
assertThat(exception)
        .hasMessageThat()
        .startsWith("Noised count must be computed before calling this function.");